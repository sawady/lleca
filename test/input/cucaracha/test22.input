
fun main() {
  z := 0
  putNum(6 * 7 + z) putChar(10)
  z := z + 1
  putNum(7 * 6 + z) putChar(10)
  x := 6
  z := z + 1
  putNum(x * 7 + z) putChar(10)
  z := z + 1
  putNum(7 * x + z) putChar(10)
  x := x + 1
  z := z + 1
  putNum(x * 6 + z) putChar(10)
  z := z + 1
  putNum(6 * x + z) putChar(10)
  z := z + 1
  putNum(x * (x - 1) + z) putChar(10)
  z := z + 1
  putNum((x - 1) * x + z) putChar(10)
  x := x - 1
  z := z + 1
  putNum(x * (x + 1) + z ) putChar(10)
  z := z + 1
  putNum((x + 1) * x + z) putChar(10)
}
