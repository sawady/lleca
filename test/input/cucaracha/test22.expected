Cons(
  Function(
    main,
    Nil,
    Unit,
    Cons(
      StmtAssign(
        z,
        ExprConstNum(
          0
        )
      ),
      Cons(
        StmtCall(
          putNum,
          Cons(
            ExprAdd(
              ExprMul(
                ExprConstNum(
                  6
                ),
                ExprConstNum(
                  7
                )
              ),
              ExprVar(
                z
              )
            ),
            Nil
          )
        ),
        Cons(
          StmtCall(
            putChar,
            Cons(
              ExprConstNum(
                10
              ),
              Nil
            )
          ),
          Cons(
            StmtAssign(
              z,
              ExprAdd(
                ExprVar(
                  z
                ),
                ExprConstNum(
                  1
                )
              )
            ),
            Cons(
              StmtCall(
                putNum,
                Cons(
                  ExprAdd(
                    ExprMul(
                      ExprConstNum(
                        7
                      ),
                      ExprConstNum(
                        6
                      )
                    ),
                    ExprVar(
                      z
                    )
                  ),
                  Nil
                )
              ),
              Cons(
                StmtCall(
                  putChar,
                  Cons(
                    ExprConstNum(
                      10
                    ),
                    Nil
                  )
                ),
                Cons(
                  StmtAssign(
                    x,
                    ExprConstNum(
                      6
                    )
                  ),
                  Cons(
                    StmtAssign(
                      z,
                      ExprAdd(
                        ExprVar(
                          z
                        ),
                        ExprConstNum(
                          1
                        )
                      )
                    ),
                    Cons(
                      StmtCall(
                        putNum,
                        Cons(
                          ExprAdd(
                            ExprMul(
                              ExprVar(
                                x
                              ),
                              ExprConstNum(
                                7
                              )
                            ),
                            ExprVar(
                              z
                            )
                          ),
                          Nil
                        )
                      ),
                      Cons(
                        StmtCall(
                          putChar,
                          Cons(
                            ExprConstNum(
                              10
                            ),
                            Nil
                          )
                        ),
                        Cons(
                          StmtAssign(
                            z,
                            ExprAdd(
                              ExprVar(
                                z
                              ),
                              ExprConstNum(
                                1
                              )
                            )
                          ),
                          Cons(
                            StmtCall(
                              putNum,
                              Cons(
                                ExprAdd(
                                  ExprMul(
                                    ExprConstNum(
                                      7
                                    ),
                                    ExprVar(
                                      x
                                    )
                                  ),
                                  ExprVar(
                                    z
                                  )
                                ),
                                Nil
                              )
                            ),
                            Cons(
                              StmtCall(
                                putChar,
                                Cons(
                                  ExprConstNum(
                                    10
                                  ),
                                  Nil
                                )
                              ),
                              Cons(
                                StmtAssign(
                                  x,
                                  ExprAdd(
                                    ExprVar(
                                      x
                                    ),
                                    ExprConstNum(
                                      1
                                    )
                                  )
                                ),
                                Cons(
                                  StmtAssign(
                                    z,
                                    ExprAdd(
                                      ExprVar(
                                        z
                                      ),
                                      ExprConstNum(
                                        1
                                      )
                                    )
                                  ),
                                  Cons(
                                    StmtCall(
                                      putNum,
                                      Cons(
                                        ExprAdd(
                                          ExprMul(
                                            ExprVar(
                                              x
                                            ),
                                            ExprConstNum(
                                              6
                                            )
                                          ),
                                          ExprVar(
                                            z
                                          )
                                        ),
                                        Nil
                                      )
                                    ),
                                    Cons(
                                      StmtCall(
                                        putChar,
                                        Cons(
                                          ExprConstNum(
                                            10
                                          ),
                                          Nil
                                        )
                                      ),
                                      Cons(
                                        StmtAssign(
                                          z,
                                          ExprAdd(
                                            ExprVar(
                                              z
                                            ),
                                            ExprConstNum(
                                              1
                                            )
                                          )
                                        ),
                                        Cons(
                                          StmtCall(
                                            putNum,
                                            Cons(
                                              ExprAdd(
                                                ExprMul(
                                                  ExprConstNum(
                                                    6
                                                  ),
                                                  ExprVar(
                                                    x
                                                  )
                                                ),
                                                ExprVar(
                                                  z
                                                )
                                              ),
                                              Nil
                                            )
                                          ),
                                          Cons(
                                            StmtCall(
                                              putChar,
                                              Cons(
                                                ExprConstNum(
                                                  10
                                                ),
                                                Nil
                                              )
                                            ),
                                            Cons(
                                              StmtAssign(
                                                z,
                                                ExprAdd(
                                                  ExprVar(
                                                    z
                                                  ),
                                                  ExprConstNum(
                                                    1
                                                  )
                                                )
                                              ),
                                              Cons(
                                                StmtCall(
                                                  putNum,
                                                  Cons(
                                                    ExprAdd(
                                                      ExprMul(
                                                        ExprVar(
                                                          x
                                                        ),
                                                        ExprSub(
                                                          ExprVar(
                                                            x
                                                          ),
                                                          ExprConstNum(
                                                            1
                                                          )
                                                        )
                                                      ),
                                                      ExprVar(
                                                        z
                                                      )
                                                    ),
                                                    Nil
                                                  )
                                                ),
                                                Cons(
                                                  StmtCall(
                                                    putChar,
                                                    Cons(
                                                      ExprConstNum(
                                                        10
                                                      ),
                                                      Nil
                                                    )
                                                  ),
                                                  Cons(
                                                    StmtAssign(
                                                      z,
                                                      ExprAdd(
                                                        ExprVar(
                                                          z
                                                        ),
                                                        ExprConstNum(
                                                          1
                                                        )
                                                      )
                                                    ),
                                                    Cons(
                                                      StmtCall(
                                                        putNum,
                                                        Cons(
                                                          ExprAdd(
                                                            ExprMul(
                                                              ExprSub(
                                                                ExprVar(
                                                                  x
                                                                ),
                                                                ExprConstNum(
                                                                  1
                                                                )
                                                              ),
                                                              ExprVar(
                                                                x
                                                              )
                                                            ),
                                                            ExprVar(
                                                              z
                                                            )
                                                          ),
                                                          Nil
                                                        )
                                                      ),
                                                      Cons(
                                                        StmtCall(
                                                          putChar,
                                                          Cons(
                                                            ExprConstNum(
                                                              10
                                                            ),
                                                            Nil
                                                          )
                                                        ),
                                                        Cons(
                                                          StmtAssign(
                                                            x,
                                                            ExprSub(
                                                              ExprVar(
                                                                x
                                                              ),
                                                              ExprConstNum(
                                                                1
                                                              )
                                                            )
                                                          ),
                                                          Cons(
                                                            StmtAssign(
                                                              z,
                                                              ExprAdd(
                                                                ExprVar(
                                                                  z
                                                                ),
                                                                ExprConstNum(
                                                                  1
                                                                )
                                                              )
                                                            ),
                                                            Cons(
                                                              StmtCall(
                                                                putNum,
                                                                Cons(
                                                                  ExprAdd(
                                                                    ExprMul(
                                                                      ExprVar(
                                                                        x
                                                                      ),
                                                                      ExprAdd(
                                                                        ExprVar(
                                                                          x
                                                                        ),
                                                                        ExprConstNum(
                                                                          1
                                                                        )
                                                                      )
                                                                    ),
                                                                    ExprVar(
                                                                      z
                                                                    )
                                                                  ),
                                                                  Nil
                                                                )
                                                              ),
                                                              Cons(
                                                                StmtCall(
                                                                  putChar,
                                                                  Cons(
                                                                    ExprConstNum(
                                                                      10
                                                                    ),
                                                                    Nil
                                                                  )
                                                                ),
                                                                Cons(
                                                                  StmtAssign(
                                                                    z,
                                                                    ExprAdd(
                                                                      ExprVar(
                                                                        z
                                                                      ),
                                                                      ExprConstNum(
                                                                        1
                                                                      )
                                                                    )
                                                                  ),
                                                                  Cons(
                                                                    StmtCall(
                                                                      putNum,
                                                                      Cons(
                                                                        ExprAdd(
                                                                          ExprMul(
                                                                            ExprAdd(
                                                                              ExprVar(
                                                                                x
                                                                              ),
                                                                              ExprConstNum(
                                                                                1
                                                                              )
                                                                            ),
                                                                            ExprVar(
                                                                              x
                                                                            )
                                                                          ),
                                                                          ExprVar(
                                                                            z
                                                                          )
                                                                        ),
                                                                        Nil
                                                                      )
                                                                    ),
                                                                    Cons(
                                                                      StmtCall(
                                                                        putChar,
                                                                        Cons(
                                                                          ExprConstNum(
                                                                            10
                                                                          ),
                                                                          Nil
                                                                        )
                                                                      ),
                                                                      Nil
                                                                    )
                                                                  )
                                                                )
                                                              )
                                                            )
                                                          )
                                                        )
                                                      )
                                                    )
                                                  )
                                                )
                                              )
                                            )
                                          )
                                        )
                                      )
                                    )
                                  )
                                )
                              )
                            )
                          )
                        )
                      )
                    )
                  )
                )
              )
            )
          )
        )
      )
    )
  ),
  Nil
)
